sys/include/net/gnrc/netif2/internal.h

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  net_gnrc_netif
+ * @internal
+ * @attention   **FOR GNRC-INTERNAL USE ONLY. PLEASE USE @ref net_gnrc_netapi
+ *              "NETAPI" (WITH THE INTERFACE'S PID) FOR EXTERNAL MANIPULATION**
+ * @{
+ *
+ * @file
+ * @brief   GNRC-internal network interface definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef NET_GNRC_NETIF2_INTERNAL_H
+#define NET_GNRC_NETIF2_INTERNAL_H
+
+#include "net/gnrc/netif2.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Message type for @ref netdev_event_t "netdev events"
+ */
+#define NETDEV_MSG_TYPE_EVENT   (0x1234)
+
+/**
+ * @brief   Acquires exclusive access to the interface
+ *
+ * @param[in] netif the network interface
+ *
+ * @internal
+ */
+static inline void gnrc_netif2_acquire(gnrc_netif2_t *netif)
+{
+    if (netif && (netif->ops)) {
```

Comment:
Strange to check for `netif->ops` here. It's not even used for acquiring a mutex. Good you leave a hint in the doc that `netif->ops != NULL` is necessary to acquire the mutex?

Reply:
`netif->ops` is NULL before the interface was initialized. If the interface is not initialized it is not necessary to lock its mutex.